src/py42/services/_connection.py

```diff
@@ -239,6 +240,12 @@ def _prepare_request(
         if u"Accept" not in headers:
             headers.update({u"Accept": u"application/json"})
         headers = _create_user_headers(headers)
+
+        _print_request(method, url, params=params, data=data)
```

Comment:
Had to shuffle some of this logic around because we wanted to first check if it `str` first (even though we know it alway is), and the logging method logs them as `dicts`, and it would error on byte arrays.

This is why I said we should assume it is always a `str` for now, because we are just going to switch to always use `json` anyway and we would not have to had to mess with some of the logging logic.

Nonetheless, this seems to be working okay.